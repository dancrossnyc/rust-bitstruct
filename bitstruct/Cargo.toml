[package]
name = "bitstruct"
version = "0.1.0"
authors = ["Dan Cross <crossd@gmail.com>"]
edition = "2018"

[dependencies]
bitstruct_derive = { version = "0.1.0", path = "../bitstruct_derive" }

[dev-dependencies]
trybuild = "1.0.42"
proptest = "1.0.0"
