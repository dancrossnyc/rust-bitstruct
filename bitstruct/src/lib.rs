pub use bitstruct_derive::bitstruct;