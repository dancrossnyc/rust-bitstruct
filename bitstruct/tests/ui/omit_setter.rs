use bitstruct::bitstruct;

bitstruct! {
    struct Reg(u32) {
        field1: u8 = 10 .. 18;

        #[bitstruct(omit_setter)]
        field2: u8 = 20 .. 22;
    }
}

fn main() {
    let _ = Reg(0)
        .with_field1(1) // field1 is fine
        .with_field2(2); // fails because omit_setter was specified on field2
}
