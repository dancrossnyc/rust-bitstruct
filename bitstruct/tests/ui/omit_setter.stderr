error[E0599]: no method named `with_field2` found for struct `Reg` in the current scope
  --> $DIR/omit_setter.rs:15:10
   |
3  | bitstruct! {
   | ---------- method `with_field2` not found for this
...
15 |         .with_field2(2); // fails because omit_setter was specified on field2
   |          ^^^^^^^^^^^ help: there is an associated function with a similar name: `with_field1`
